modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2353,6 +2384,15 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(
                 }), ioExecutor);
     }
 
+    /**
+     * For HA zones: Check that last rebalance was graceful (raft group maintained the majority) rather than forced (caused by a disaster
```

Comment:
```suggestion
     * For HA zones: Check that last rebalance was graceful (caused by common rebalance triggers, like data nodes change, replica factor change, etc.) rather than forced (caused by a disaster
```